airflow/utils/context.py

```diff
@@ -274,14 +279,27 @@ def __getitem__(self, key: int | str | Asset | AssetAlias) -> LazyAssetEventSele
             obj = self._inlets[key]
             if not isinstance(obj, (Asset, AssetAlias)):
                 raise IndexError(key)
+
+            if isinstance(obj, AssetRef):
+                obj = _fetch_active_assets_by_name([obj.name], self._session)
```

Comment:
Is this correct? An IndexError is raised when `obj` is _not_ Asset or AssetAlias; how can it reach the next if below?

Reply:
just rewrote it